### PR TITLE
Update program input section for getting started guides for sampler and estimator

### LINKED ISSUE #288: backend specification is not optional
**Describe the bug**
https://github.com/Qiskit/qiskit-ibm-runtime/blob/main/docs/tutorials/how-to-getting-started-with-sampler.ipynb says:
The Sampler takes in:
[...]
- **Optional**: The **backend** to run on. If one is not specified, the least busy backend is used.

However, not specifying a backend yields an error

**Steps to reproduce**
```
with Sampler(circuits=bell, service=service) as sampler:
result = sampler(circuit_indices=[0], shots=1024)
print(result)
```

--> `IBMInputValueError: '"backend_name" is required field in "options" for ``ibm_quantum`` runtime.'`

**Expected behavior**
run on least busy backend

**Suggested solutions**
make backend_name optional as described or change relevant tutorial text. 

**Additional Information**

- **qiskit-ibm-runtime version**: 0.4.0
- **Python version**: 3.8.2
- **Operating system**: Windows 10

### PR DESCRIPTION
**Pull request by HuangJunye**
### Summary

1. Updated getting started guides for sampler and estimator to reflect that backend is required for IBM Quantum while it's optional for IBM Cloud. 
2. Changed the format of program inputs to be closer to the docstring
3. Add link to the respective API reference for Sampler and Estimator.

### Details and comments
Fixes #288

### CODE REVIEW

File: docs/tutorials/how-to-getting-started-with-estimator.ipynb [hunk 1]
```diff
@@ -31,13 +31,16 @@
...
-    "\n",
-    "- The **circuits** you want to investigate.\n",
-    "- The **parameters** input to evaluate the circuits.\n",
-    "- The **observables** (Hamiltonians) to be evaluated.\n",
-    "- Optional: The **backend** to run on. If one is not specified, the least busy backend is used.\n",
-    "- Optional: The instruction to **skip_transpilation**.\n",
+    "The Estimator takes in the following arguments:\n",
+    "- **circuits**: a list of (parameterized) circuits that you want to investigate.\n",
+    "- **observables**: a list of observables to measure the expectation values.\n",
+    "- **parameters**: a list of parameters for the parameterized circuits. It should be ommitted if the circuits provided are not parameterized.\n",
```
Comment (rathishcholarajan) [inline docs/tutorials/how-to-getting-started-with-estimator.ipynb:37]:
```suggestion
    "- **parameters**: a list of parameters for the parameterized circuits. It should be omitted if the circuits provided are not parameterized.\n",
```

File: docs/tutorials/how-to-getting-started-with-estimator.ipynb [hunk 2]
```diff
@@ -31,13 +31,16 @@
...
+    "- **skip_transpilation**: circuit transpilation is skipped if set to `True`. Default value is `False`.\n",
+    "- **service**: the `QiskitRuntimeService` instance to run the program on. If not specified, the default saved account for `QiskitRuntimeServie` is initialized.\n",
```
Comment (rathishcholarajan) [inline docs/tutorials/how-to-getting-started-with-estimator.ipynb:39]:
```suggestion
    "- **service**: the `QiskitRuntimeService` instance to run the program on. If not specified, the default saved account for `QiskitRuntimeService` is initialized.\n",
```

File: docs/tutorials/how-to-getting-started-with-sampler.ipynb [hunk 3]
```diff
@@ -28,12 +28,15 @@
...
     "\n",
-    "The Sampler takes in:\n",
-    "\n",
-    "- The **circuits** you want to investigate.\n",
-    "- The **parameters** input to evaluate the circuits.\n",
-    "- Optional: The **backend** to run on. If one is not specified, the least busy backend is used.\n",
-    "- Optional: The instruction to **skip_transpilation**.\n",
+    "The Sampler takes in the following arguments:\n",
+    "- **circuits**: a list of (parameterized) circuits that you want to investigate.\n",
+    "- **parameters**: a list of parameters for the parameterized circuits. It should be ommitted if the circuits provided are not parameterized.\n",
```
Comment (rathishcholarajan) [inline docs/tutorials/how-to-getting-started-with-sampler.ipynb:33]:
```suggestion
    "- **parameters**: a list of parameters for the parameterized circuits. It should be omitted if the circuits provided are not parameterized.\n",
```

File: docs/tutorials/how-to-getting-started-with-sampler.ipynb [hunk 4]
```diff
@@ -28,12 +28,15 @@
...
+    "- **skip_transpilation**: circuit transpilation is skipped if set to `True`. Default value is `False`.\n",
+    "- **service**: the `QiskitRuntimeService` instance to run the program on. If not specified, the default saved account for `QiskitRuntimeServie` is initialized.\n",
```
Comment (rathishcholarajan) [inline docs/tutorials/how-to-getting-started-with-sampler.ipynb:35]:
```suggestion
    "- **service**: the `QiskitRuntimeService` instance to run the program on. If not specified, the default saved account for `QiskitRuntimeService` is initialized.\n",
```

**Review by rathishcholarajan — CHANGES_REQUESTED** [review]
Looks good except for some typos...

**Comment by HuangJunye** [comment]
@rathishcholarajan Thanks for spotting the typos! I should install a typo check plugin in my jupyter lab...

**Review by rathishcholarajan — APPROVED** [review]
LGTM!

**Comment by HuangJunye** [comment]
@rathishcholarajan Seems the doc was not triggered to build, should I start it? Is this the workflow?
image

**Comment by rathishcholarajan** [comment]
Yes this is the one, I just ran it this morning. Only thing we need to be careful is to not release docs for any unreleased feature. Right now we haven't merged any features after 0.4, so it's fine for now.

**Pull request merged**